tests/test_issue_8953_zsh_prompt_wrap_guard.py

```
@@ -0,0 +1,257 @@
+            chunk = os.read(master_fd, 65_536)
+        except OSError:
+            master_open = False
+            continue
+        if chunk:
+            output.extend(chunk)
+        else:
+            master_open = False
+
+    os.kill(pid, signal.SIGKILL)
+    os.waitpid(pid, 0)
+    raise AssertionError(f"interactive zsh did not exit; output={bytes(output)!r}")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)